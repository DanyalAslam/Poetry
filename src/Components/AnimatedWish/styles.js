
import { Dimensions, Platform } from 'react-native'

const vw = Dimensions.get('window').width * 0.01
const vh = Dimensions.get('window').height * 0.01

const styles = {
    wishIcon: {
        height: 30 * vw,
        width: 30 * vw, 
        top: -3.65 * vh,
        right: 6 * vw,
        zIndex: 10000,

    },
    wishImageCon: {
        position: 'absolute',
        top: 3 * vh,
        right: 4 * vw,
        zIndex: 9999999, 
        elevation: 5,
        height: 10 * vw,
        width: 10 * vw,
        ...Platform.select({
            ios:{
                top: -13 * vh,
                right: -4 * vw,
            }
        })
    },
}
export default styles