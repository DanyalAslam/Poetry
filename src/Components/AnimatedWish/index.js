import React, { Component } from 'react'
import {
    TouchableOpacity,
    Animated, Easing,
    View,
} from 'react-native'

import HeartFill from './heart.json';
import LottieView from 'lottie-react-native';
import styles from './styles'


export default class AnimatedWish extends Component {
    constructor(props) {
        super(props)
        this.progress = new Animated.Value(0);
    }

    render() {
 

        return (
            // <View style={{ flex: 1, }}>

                <TouchableOpacity style={[styles.wishImageCon,this.props.style]}

                    onPress={this.props.onWishPress}>
                    {
                        (this.props.wish == 'wish')
                            ? Animated.timing(this.progress, {
                                toValue: 0.9,
                                duration: 1000,
                                easing: Easing.linear,
                                useNativeDriver: true
                            }).start()
                            : (this.props.wish == 'unwish')
                                ? Animated.timing(this.progress, {
                                    toValue: 0,
                                    duration: 1000,
                                    easing: Easing.linear,
                                    useNativeDriver: true
                                }).start()
                                : null

                    }

                    {
                        this.props.wish
                            ?
                            <LottieView
                                ref={animation => {
                                    this.animation = animation;
                                }}
                                source={HeartFill}
                                progress={this.progress}
                                style={styles.wishIcon}
                            />
                            : null
                    }


                </TouchableOpacity>
            // </View>
        )

    }
}