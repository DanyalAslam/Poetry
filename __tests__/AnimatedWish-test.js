/**
 * @format
 */

import 'react-native';
import React from 'react';
import AnimatedWish from '../src/Components/AnimatedWish';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';


test('renders correctly', () => {
    const tree = renderer.create(<AnimatedWish />).toJSON();
    expect(tree).toMatchSnapshot();
  });

