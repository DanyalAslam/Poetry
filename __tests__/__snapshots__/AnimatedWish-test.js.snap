// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "elevation": 5,
      "height": 75,
      "opacity": 1,
      "position": "absolute",
      "right": -30,
      "top": -173.42,
      "width": 75,
      "zIndex": 9999999,
    }
  }
/>
`;
